src/lib/common/analyser.py

```diff
@@ -206,12 +206,17 @@ def _run(self, config):
 
             self.analyse_element(element, config)
 
+        self.post_analyse(config)
         save_logs(self.__logs, self.ANALYSER_LOGS)
 
-    def setup_run(self):
+    def setup_run(self, config):
```

Comment:
this should be renamed to just `setup`, I think.

Reply:
Or maybe `pre_analyse`?

Reply:
I went with pre_/post_analyse